modules/nosql/manager_redis.go

```diff
@@ -59,8 +60,84 @@ func (m *Manager) GetRedisClient(connection string) redis.UniversalClient {
 		name: []string{connection, uri.String()},
 	}
 
+	opts := getRedisOptions(uri)
+	tlsConfig := getRedisTLSOptions(uri)
+
+	clientName := uri.Query()["clientname"]
+
+	if clientName != nil && len(clientName) > 0 {
+		client.name = append(client.name, clientName[0])
+	}
+
+	switch uri.Scheme {
+	case "redis+sentinels":
+		fallthrough
+	case "rediss+sentinel":
+		opts.TLSConfig = tlsConfig
+		fallthrough
+	case "redis+sentinel":
+		if uri.Host != "" {
+			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
+		}
+		if uri.Path != "" {
+			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
+				opts.DB = db
+			}
+		}
+
+		client.UniversalClient = redis.NewFailoverClient(opts.Failover())
+	case "redis+clusters":
+		fallthrough
+	case "rediss+cluster":
+		opts.TLSConfig = tlsConfig
+		fallthrough
+	case "redis+cluster":
+		if uri.Host != "" {
+			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
+		}
+		if uri.Path != "" {
+			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
+				opts.DB = db
+			}
+		}
+		client.UniversalClient = redis.NewClusterClient(opts.Cluster())
+	case "redis+socket":
+		simpleOpts := opts.Simple()
+		simpleOpts.Network = "unix"
+		simpleOpts.Addr = path.Join(uri.Host, uri.Path)
+		client.UniversalClient = redis.NewClient(simpleOpts)
+	case "rediss":
+		opts.TLSConfig = tlsConfig
+		fallthrough
+	case "redis":
+		if uri.Host != "" {
+			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
+		}
+		if uri.Path != "" {
+			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
+				opts.DB = db
+			}
+		}
+		client.UniversalClient = redis.NewClient(opts.Simple())
+	default:
+		return nil
```

Comment:
Adding a error log would help users to find their mistakes ASAP

Reply:
Same as the other comments!